packages/@webex/plugin-meetings/test/unit/spec/meeting/muteState.js

```diff
@@ -509,3 +509,221 @@ describe('plugin-meetings', () => {
     });
   });
 });
+
+describe('#init, #handleLocalTrackChange', () => {
+  let meeting;
+  let muteState;
+  let originalRemoteUpdateAudioVideo;
+  let applyUnmuteAllowedToTrackSpy, muteLocalTrackSpy, applyClientStateToServerSpy, setServerMutedSpy;
+  let setMutedSpy;
+  const fakeLocus = {info: 'this is a fake locus'};
+
+  const createFakeLocalTrack = (id, muted) => {
+    return {
+      id,
+      setMuted: sinon.stub(),
+      setServerMuted: sinon.stub(),
+      setUnmuteAllowed: sinon.stub(),
+      muted,
+    };
+  };
+
+  const setup = (mediaType, remoteMuted = false, muted = false, defineTracks = true) => {
+
+    const remoteMuteField = mediaType === AUDIO ? 'remoteMuted' : 'remoteVideoMuted';
+
+    meeting = {
+      mediaProperties: {
+        audioTrack: defineTracks ? createFakeLocalTrack('fake audio track', muted) : undefined,
+        videoTrack: defineTracks ? createFakeLocalTrack('fake video track', muted) : undefined,
+      },
+      [remoteMuteField]: remoteMuted,
+      unmuteAllowed: true,
+      unmuteVideoAllowed: true,
+
+      locusInfo: {
+        onFullLocus: sinon.stub(),
+      },
+      members: {
+        selfId: 'fake self id',
+        muteMember: sinon.stub().resolves(),
+      },
+    };
+
+    const direction = mediaType === AUDIO ? {sendAudio: true} : {sendVideo: true};
+    muteState = createMuteState(mediaType, meeting, direction, false);
+
+    originalRemoteUpdateAudioVideo = MeetingUtil.remoteUpdateAudioVideo;
+
+    MeetingUtil.remoteUpdateAudioVideo = sinon.stub().resolves(fakeLocus);
+  }
+
+   const setupSpies = (muteState, mediaType) => {
+    applyUnmuteAllowedToTrackSpy = sinon.spy(muteState, 'applyUnmuteAllowedToTrack');
```

Comment:
applyUnmuteAllowedToTrack, muteLocalTrack and applyClientStateToServer are all private functions. I don't think we should be spying on them - this makes the tests very closely coupled with the implementation, instead we should be mocking/spying the external methods/functions that MuteState calls, so for example instead of muteLocalTrack we should spy on the setServerMuted() method of the right track, instead of applyUnmuteAllowedToTrack we should spy on setUnmuteAllowed() on the right track, instead of applyClientStateToServer we should be checking that the right requests are sent to the server